lib/src/lib/tokenizer/charTokenizer.ts

```diff
@@ -2,17 +2,17 @@ import { Region } from "@dodona/dolos-core";
 import { Token, Tokenizer } from "./tokenizer.js";
 
 export class CharTokenizer extends Tokenizer {
-  /**
-   * Runs the parser on a given string. Returns an async iterator returning
-   * tuples containing the stringified version of the token and the
-   * corresponding position.
-   *
-   * @param text The text string to parse
-   */
-  public *generateTokens(text: string): IterableIterator<Token> {
-    for (const [lineNumber, line] of text.split("\n").entries())
-      yield* line
-        .split("")
-        .map((char, col) => this.newToken(char, new Region(lineNumber, col, lineNumber, col + 1)));
+
+  generateTokens(text: string): Token[] {
+    const tokens = [];
+    for (const [lineNumber, line] of text.split("\n").entries()) {
+      tokens.push(
+        line
+          .split("")
+          .map((char, col) => this.newToken(char, new Region(lineNumber, col, lineNumber, col + 1)))
+      );
+    }
+
+    return tokens.flat();
```

Comment:
(sorry, I don't want to nitpick on code/micro manage you. I just checked this PR out of interest and saw a potential optimization)

I don't know if this code is performance critical, but it might be improved by:
- iterating over the line using a simple loop instead of `split("")` will cause a lot less overhead, no new array needs to be created, etc.
- by adding rewriting it as a loop, I think you can add the stuff directly to tokens, avoiding `flat`

There might of course be reasons why the code was written like this that I'm not aware of.

Reply:
> I don't know if this code is performance critical

This code will only be used when you try to tokenize a `.txt` or `.md` file. Thus I did not really look at ways to improve this code and just adjusted it to return a list instead of an iterator. This to meet the return type specified by the abstract class.

But my introduced `.flat()` is indeed totally not necessary here. I will change this!